### DISCUSSION TITLE
Lesson 4: ValueError: {'message': "sender doesn't have enough funds to send tx.

### DISCUSSION THREAD

**Discussion by karlMoreno** [smartcontractkit/full-blockchain-solidity-course-py#764] [Q&A]
Hi guys! I've been on this issue for a couple days and nothing I try seems to work. I have all the right code included below but I've hit a wall at line 71.

tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)

and it gives me the error:
Deploying Contract!
Traceback (most recent call last):
  File "deploy.py", line 71, in <module>
    tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
  File "/usr/local/lib/python3.8/site-packages/web3/eth.py", line 698, in send_raw_transaction
    return self._send_raw_transaction(transaction)
  File "/usr/local/lib/python3.8/site-packages/web3/module.py", line 57, in caller
    result = w3.manager.request_blocking(method_str,
  File "/usr/local/lib/python3.8/site-packages/web3/manager.py", line 187, in request_blocking
    return self.formatted_response(response,
  File "/usr/local/lib/python3.8/site-packages/web3/manager.py", line 168, in formatted_response
    raise ValueError(response["error"])
ValueError: {'message': "sender doesn't have enough funds to send tx. The upfront cost is: 7525300000000000 and the sender's account only has: 0", 'code': -32000, 'data': {'stack': "Error: sender doesn't have enough funds to send tx. The upfront cost is: 7525300000000000 and the sender's account only has: 0\n    at VM.<anonymous> (/Applications/Ganache.app/Contents/Resources/static/node/node_modules/ganache-core/node_modules/ethereumjs-vm/lib/runTx.ts:114:11)\n    at step (/Applications/Ganache.app/Contents/Resources/static/node/node_modules/ganache-core/node_modules/ethereumjs-vm/dist/runTx.js:33:23)\n    at Object.next (/Applications/Ganache.app/Contents/Resources/static/node/node_modules/ganache-core/node_modules/ethereumjs-vm/dist/runTx.js:14:53)\n    at fulfilled (/Applications/Ganache.app/Contents/Resources/static/node/node_modules/ganache-core/node_modules/ethereumjs-vm/dist/runTx.js:5:58)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)", 'name': 'Error'}}

Thanks for looking at my issue and hope to hear from you guys soon!
![Screen Shot 2022-01-12 at 1 14 51 AM](https://user-images.githubusercontent.com/86685561/149098730-7d262f86-2aaf-41ef-adc6-66735e6c23f1.png)

**Comment by cromewar** [comment]
Hello @karlMoreno make sure your private key is correctly imported from the .env variables, and also make sure you have enough Eth on your wallet to pass the transaction, you said you have the exact same code as on the video, just to verify this,  could you please clone the repo and try to run it to see if it works.
